src/index.d.ts

```diff
@@ -114,15 +115,23 @@ export type TVCompoundSlots<
       } & ClassProp
 >;
 
-export type TVDefaultVariants<V extends TVVariants<S>, EV extends TVVariants, S extends TVSlots> = {
-  [K in keyof V | keyof EV]?: StringToBoolean<keyof V[K] | keyof EV[K]>;
+export type TVDefaultVariants<
+  V extends TVVariants<S>,
+  S extends TVSlots,
+  EV extends TVVariants<ES>,
+  ES extends TVSlots,
+> = {
+  [K in keyof V | keyof EV]?:
+    | (K extends keyof V ? StringToBoolean<keyof V[K]> : never)
+    | (K extends keyof EV ? StringToBoolean<keyof EV[K]> : never);
```

Comment:
Had to split this into two separate pieces otherwise it would error since keys from both `K` and `EV` can't index `V` or `EV`